### PR TITLE
race was fixed in k8s 1.18

### PR DESCRIPTION
**Pull request by dprotaso**
See discussion: https://github.com/kubernetes/kubernetes/issues/90952

### CODE REVIEW

File: leaderelection/context_test.go [hunk 1]
```diff
@@ -1,6 +1,3 @@
-// +build !race
-// TODO(https://github.com/kubernetes/kubernetes/issues/90952): Remove the above.
```
Comment (mattmoor) [inline leaderelection/context_test.go:2]: We should delete this too: https://github.com/knative/pkg/blob/ed076316cbe1107590822f930f4ed20b109290f1/test/presubmit-tests.sh#L48

**Review by vagababov — COMMENTED** [review]
/lgtm
/approve

File: leaderelection/context_test.go (hunk 1 above)
Comment (mattmoor) [reply leaderelection/context_test.go:2]: @vagababov 👀

Comment (vagababov) [reply leaderelection/context_test.go:2]: Sorry :| 
Somehow I missed this.
😢

**Pull request merged**